### PR TITLE
Move GET api requests api/v1/maps

### LINKED ISSUE #3570: Move GET api requests api/v1/maps
includes both `/api/v1/maps` and `api/v1/maps/:map_id/layers` GET endpoints

branch: `3570-new-api-maps-get-endpoints`

feat. flag: `active_record_layers_endpoint`

**Warning**: This branch has merged `3535-new-users-layers-endpoints` to reuse work already done there.

### PR DESCRIPTION
**Pull request by Kartones**
Closes #3570 

**Don't merge before #3568**

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower CR please (probably has lots of code duplicated from [here](https://github.com/CartoDB/cartodb/pull/3568) but I needed some logic and FE blockade forbid me from releasing this branch...

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3593/))

File: app/controllers/carto/api/maps_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      def load_map
+        # User must be owner or have permissions for the map's visualization
+        vis = Carto::Visualization.where({
+            user_id: current_user.id,
+            map_id: params[:id],
+            kind: Carto::Visualization::KIND_GEOM
+          }).first
+        raise RecordNotFound if vis.nil?
+
+        @map = Carto::Map.where(id: params[:id]).first
```
Comment (rafatower) [inline app/controllers/carto/api/maps_controller.rb]: It's the controller's responsiblity to ensure it is receiving a valid `id`

**Comment by rafatower** [comment]
Aside from the only comment and whitespaces review, looks ok :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3600/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3601/))

**Pull request merged**